dmcp/fix.py

```diff
@@ -12,45 +13,59 @@
 from cvxpy.constraints.zero import Zero
 from cvxpy.constraints.psd import PSD
 
+from cvxpy.constraints.nonpos import NonNeg
+from cvxpy.constraints.nonpos import Inequality
+
 def fix(obj, fix_vars):
     """
     Fix the given variables in the object
     :param obj: a problem or an expression
     :param fix_var: a list of variables
     :return: a problem or an expression
     """
+    # Test whether fix_vars is already list, if not, recast
+    if not isinstance(fix_vars,list):
+       fix_vars = [fix_vars]
 
     # Create list of parameters
     variable_list = obj.variables()
     variable_list.sort(key = lambda x:x.id)
     param_list = []
-    for var in variable_list:
+    for var in variable_list:        
         if var.sign == "NONNEGATIVE":
             para = cvx.Parameter(shape = var.shape, nonneg=True)
             if var.value is not None:
                 para.value = abs(var.value)
+            else:
+                para.value = np.zeros(var.shape)
             para.id = var.id
             param_list.append(para)
         elif var.sign == "NONPOSITIVE":
             para = cvx.Parameter(shape = var.shape, nonpos=True)
             if var.value is not None:
                 para.value = -abs(var.value)
+            else:
+                para.value = np.zeros(var.shape)
             para.id = var.id
             param_list.append(para)
         elif var.attributes['PSD'] == True:
             para = cvx.Parameter(shape = var.shape, PSD=True)
             if var.value is not None:
                 para.value = var.value
+            else:
+                para.value = np.zeros(var.shape)
             para.id = var.id
             param_list.append(para)
         else:
             para = cvx.Parameter(shape = var.shape)
             if var.value is not None:
                 para.value = var.value
+            else:
```

Comment:
Can you group all the None cases together?

Reply:
sg